Add Android modules to dependencies and topology.

srmurguia opened the pull request:
Now it is possible to define dependencies from and to Android modules,
also, all the topology is now applied to Android modules too.

srmurguia committed:
Add Android modules to dependencies and topology.

Now it is possible to define dependencies from and to Android modules,
also, all the topology is now applied to Android modules too.

NikitaKozlov commented on src/main/kotlin/com/google/androidstudiopoet/DependencyValidator.kt:
```diff
@@ -19,13 +19,64 @@ package com.google.androidstudiopoet
 class DependencyValidator {
-    fun isValid(dependencies: List<DependencyConfig>, moduleCount: Int): Boolean {
+
+    private val ANDROID_TYPE = "androidAppModule"
```
make both of them `const`, please.

NikitaKozlov commented on src/main/kotlin/com/google/androidstudiopoet/DependencyValidator.kt:
```diff
@@ -19,13 +19,64 @@ package com.google.androidstudiopoet
+        return true
+    }
+
+    private class ModuleSplit(moduleName : String){
```
Can you please move the class to the end of the file?

NikitaKozlov commented on src/main/kotlin/com/google/androidstudiopoet/DependencyValidator.kt:
```diff
@@ -19,13 +19,64 @@ package com.google.androidstudiopoet
+            return true
+        }
+        // check that a java module does not depend on an Android module
+        if (from.type.equals(MODULE_TYPE) && to.type.equals(ANDROID_TYPE)) {
```
Please, instead of comment make a method with proper name. For the next `if` as well. Then it would be also possible to have one `when` instead of 5 `if`, but it is optional, if you like `if`s keep it like it is.

srmurguia replied:
Thanks for the comments! I have modified the code to address those issues and now the file looks cleaner! Please, take a look.

srmurguia committed:
Add feedback from Nikita on DependencyValidator.kt:

- Make mdoule types constants
- Move ModuleSplit to end of file
- Create a couple of functions that check some dependency rules instead
of having the checks directly on trhe function that uses them.

All those changes made the code cleaner.
